gemini-dsl/src/main/java/com/haroncode/gemini/dsl/ConnectionFactoryDsl.kt

```diff
@@ -0,0 +1,37 @@
+package com.haroncode.gemini.dsl
+
+import com.haroncode.gemini.connection.AutoStoreDisposeConnectionRule
+import com.haroncode.gemini.connection.ConnectionRule
+import com.haroncode.gemini.core.Store
+
+/**
+ * @author HaronCode.
+ */
+inline fun <reified T : Any> connectionFactory(
+    crossinline factory: ConnectionRuleListBuilder.(param: T) -> Unit
+): ConnectionRule.Factory<T> = object : ConnectionRule.Factory<T> {
+
+    override fun create(param: T): Collection<ConnectionRule> {
+        val builder = ConnectionRuleListBuilder()
+        builder.factory(param)
+        return builder.build()
+    }
+}
+
+class ConnectionRuleListBuilder {
+
+    private val mutableList = mutableListOf<ConnectionRule>()
+
+    fun connection(connectionRule: () -> ConnectionRule) {
+        connectionRule.invoke()
+            .let(mutableList::add)
+    }
+
+    fun autoDispose(storeFactory: () -> Store<*, *, *>) {
```

Comment:
maybe storeProvider? Provider is more abstract than factory, like in Dagger

Reply:
done